File: eval/code_corpus/safe/xss.py
"""SAFE variant of xss.py — output is HTML-escaped before reflection.

Negative case: the eval expects ZERO findings for this file.
"""
from html import escape

from flask import request


def greet():
    name = request.args.get("name", "")
    # SAFE: escape() neutralizes <, >, & before it reaches the page.
    return "<h1>Hello " + escape(name) + "</h1>"

File: eval/code_corpus/safe/xss.js
// SAFE variant of xss.js — textContent never parses markup.
// Negative case: the eval expects ZERO findings for this file.

function showName(name) {
  const el = document.getElementById("greeting");
  // SAFE: textContent treats the value as text, not HTML — no script execution.
  el.textContent = "Hello " + name;
}

module.exports = { showName };
